subsys/fs/shell.c

```diff
@@ -36,7 +36,7 @@ static struct fs_mount_t fatfs_mnt = {
 
 /* TODO: Implement dynamic storage dev selection */
 #ifdef CONFIG_FS_LITTLEFS_BLK_DEV
-FS_LITTLEFS_DECLARE_DEFAULT_CONFIG(lfs_data);
+static struct fs_littlefs lfs_data = {0};
```

Comment:
mount of littlefs for blk device fails in case when we use `FS_LITTLEFS_DECLARE_DEFAULT_CONFIG` initializer
make the same initializator as for little fs sample

Reply:
As I can see FS_LITTLEFS_DECLARE_DEFAULT_CONFIG creates not only fs_littlefs structure, but read_buffer, prog_buffer and lookahead_buffer and cachesize = 0.  FS_LITTLEFS_DECLARE_CUSTOM_CONFIG comment says that application can define and initialize struct fs_littlefs directly, but application is responsible to ensure the configuration. If I understood correctly you had to set CONFIG_FS_LITTLEFS_FC_HEAP_SIZE in config because you are using cachesize = 0 in lfs_data. I'm not sure if it's correct. As for me - I suggest to make 2 things: 1) Make lfs_data initialization for each driver (move it under ifdef) and use FS_LITTLEFS_DECLARE_CUSTOM_CONFIG to set correct values. If it's impossible and lfs_data should be 0 for both drivers - write a comment with the desctiption why it's so.

Reply:
I just copied and pasted parts of code from LittleFS samples for block devices and haven't delved deep into it, it works correctly. The problem was with the `FS_LITTLEFS_DECLARE_DEFAULT_CONFIG` --- the LittleFS mount call getting stuck. I'll investigate your concerns.

Reply:
changed to ` FS_LITTLEFS_DECLARE_CUSTOM_CONFIG` and add description to commit message why it helps

Reply:
+1